### PR TITLE
Exit 1 if conclusion is failure and -fail-on-error for github-check reporter

### PR DESCRIPTION
**Pull request by haya14busa**
For #446 and #403

### CODE REVIEW

File: reviewdog.go [hunk 1]
```diff
@@ -11,6 +11,8 @@ import (
 	"github.com/reviewdog/reviewdog/difffilter"
 )
 
+var X = 14
```
Comment (review-dog) [inline reviewdog.go:14]: **[golint-circleci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
exported var X should have comment or be unexported

**Comment by haya14busa** [comment]
image

**Comment by GieForce** [comment]
Hi @haya14busa, 

I'm currently trying to set up a git hook with `-fail-on-error` flag for my local environment. 
For platforms like Github it's working fine by having this command: 
`reviewdog -reporter=github-check -fail-on-error -filter-mode=nofilter` 
However, there seems no option to execute the same on fail-on-error for the local reporter. 
Can this be added to reviewdog? Or is there some other way to reach the same result. 
I already tried to add like how it's done on this [git-hook](https://github.com/reviewdog/reviewdog/tree/master/.git_hooks) but I cannot get this working

**Comment by haya14busa** [comment]
It should work for local reported as well.

**Comment by GieForce** [comment]
I realized after reading the date of this pull request that my version was not up to date. 
Sorry for the inconvenience.

**Pull request merged**